pkg/startup/startup.go

```diff
@@ -244,6 +222,61 @@ func Run() {
 		log.WithError(err).Errorf("Unable to ensure network for os")
 		terminate()
 	}
+
+	if !runOnce {
```

Comment:
Looking at this, I'm starting to think that this belongs separately from `startup.go`. 

First thing I'm thinking is that it doesn't make sense to have this code run multiple times:
- https://github.com/projectcalico/node/blob/87af881af17d24f785309cc43883b59778d51c85/pkg/startup/startup.go#L94-L118

And also this code should only need to run at startup, and will never need to be periodic:
- https://github.com/projectcalico/node/blob/87af881af17d24f785309cc43883b59778d51c85/pkg/startup/startup.go#L124-L170

Configuring the AS number will never change, since it's done based on an env var so needs a pod restart anyway:
- https://github.com/projectcalico/node/blob/87af881af17d24f785309cc43883b59778d51c85/pkg/startup/startup.go#L183

And same for configuring IP pools:
- https://github.com/projectcalico/node/blob/87af881af17d24f785309cc43883b59778d51c85/pkg/startup/startup.go#L207-L224

Now, I understand that this PR doesn't include those in the periodic / poll loop. But, it does introduce a behavior where all of that now gets executed _twice_ per node launch (once in the initial call to startup, and once when the startup process is launched).

I think we should have a new routine for the periodic code so that it doesn't call all of that again (and we can name it something other than startup, since it's not just at startup anymore). Most of what you've done here would be the same, but we'd want to commonize/export the IP auto-detection function so it could be used in both places. WDYT?